CentralHub.Api/Services/LocalizationTargetService.cs

```diff
@@ -0,0 +1,34 @@
+using CentralHub.Api.Model;
+
+namespace CentralHub.Api.Services;
+
+public class LocalizationTargetService : ILocalizationTargetService
+{
+    private Dictionary<int, List<Measurement>> Measurements = new Dictionary<int, List<Measurement>>();
+
+    public void AddMeasurements(int id, List<Measurement> measurements)
+    {
+        lock (Measurements)
+        {
+            if (Measurements.ContainsKey(id))
+            {
+                Measurements[id].AddRange(measurements);
+            }
+            else
+            {
+                Measurements.Add(id, measurements);
```

Comment:
You should make a copy of measurements because the user can keep measurements around and that is bad because now other threads can access this list